packages/ketcher-core/src/domain/entities/sgroup.ts

```diff
@@ -596,6 +598,58 @@ export class SGroup {
     }
     return c
   }
+
+  static isAtomInContractedSGroup = (atom, sGroups) => {
+    const contractedSGroup: number[] = []
+
+    sGroups.forEach((sGroup) => {
+      const id = 'item' in sGroup ? sGroup.item.id : sGroup.id
+      if (this.isContractedSGroups(id, sGroups)) {
+        contractedSGroup.push(id)
+      }
+    })
+    return contractedSGroup.some((sg) => atom.sgs.has(sg))
+  }
+
+  static isBondInContractedSGroup(
+    bond: Bond,
+    sGroups: Map<number, ReSGroup> | Pool<SGroup>
+  ) {
+    return [...sGroups.values()].some((sGroup) => {
+      const atomsInSGroup = 'item' in sGroup ? sGroup.item.atoms : sGroup.atoms
+      const isContracted = !this.isExpandedSGroup(sGroup)
+      return (
+        isContracted &&
+        atomsInSGroup.includes(bond?.begin) &&
+        atomsInSGroup.includes(bond?.end)
+      )
+    })
+  }
+
+  static isSuperAtom(sGroup) {
+    if (!sGroup) {
+      return false
+    }
+    return sGroup.type === 'SUP'
+  }
+
+  static isExpandedSGroup(sGroup) {
+    return 'item' in sGroup ? sGroup?.item.data.expanded : sGroup.data.expanded
```

Comment:
Don't know if we need to add this now, but in future it would be more convenient if we had
```
static isExpandedSGroup(sGroup: SGroup | ReSgroup) {
  if (SGroup.isSGroup(sGroup)) {
    return 'item' in sGroup ? sGroup?.item.data.expanded : sGroup.data.expanded
  }
}
```
May be this can be used: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates